From 1cb583416142e13bea86df69a148dbd034a78e36 Mon Sep 17 00:00:00 2001
From: Ryan Blue <blue@apache.org>
Date: Wed, 1 Jul 2015 16:46:23 -0700
Subject: [PATCH 144/176] PARQUET-325: Always use row group size when padding
 is 0.

For block file systems, if the size left in the block is greater than
the max padding, a row group will be targeted at the remaining size.
However, when using 0 to turn padding off, the remaining bytes will
always be greater than padding and row groups can be targeted at very
tiny spaces. When padding is off, the next row group's size should
always be the default size.

Author: Ryan Blue <blue@apache.org>

Closes #234 from rdblue/PARQUET-325-padding-0-fix and squashes the following commits:

f4b3c2b [Ryan Blue] PARQUET-325: Always use row group size when padding is 0.
---
 .../java/parquet/hadoop/ParquetFileWriter.java     |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/parquet-hadoop/src/main/java/parquet/hadoop/ParquetFileWriter.java b/parquet-hadoop/src/main/java/parquet/hadoop/ParquetFileWriter.java
index 7c83f4d..38a44c2 100644
--- a/parquet-hadoop/src/main/java/parquet/hadoop/ParquetFileWriter.java
+++ b/parquet-hadoop/src/main/java/parquet/hadoop/ParquetFileWriter.java
@@ -665,6 +665,10 @@ public class ParquetFileWriter {
 
     @Override
     public long nextRowGroupSize(FSDataOutputStream out) throws IOException {
+      if (maxPaddingSize <= 0) {
+        return rowGroupSize;
+      }
+
       long remaining = dfsBlockSize - (out.getPos() % dfsBlockSize);
 
       if (isPaddingNeeded(remaining)) {
-- 
1.7.9.5

